SU2_CFD/src/numerics_direct_mean.cpp

```diff
@@ -3274,9 +3274,11 @@ void CUpwRoe_Flow::ComputeResidual(su2double *val_residual, su2double **val_Jaco
   if (RoeSoundSpeed2 <= 0.0) {
     for (iVar = 0; iVar < nVar; iVar++) {
       val_residual[iVar] = 0.0;
-      for (jVar = 0; jVar < nVar; jVar++) {
-        val_Jacobian_i[iVar][iVar] = 0.0;
-        val_Jacobian_j[iVar][iVar] = 0.0;
+      if (implicit){
+        for (jVar = 0; jVar < nVar; jVar++) {
+          val_Jacobian_i[iVar][iVar] = 0.0;
```

Comment:
Why do we loop through jVar but access only the diagonal here? This is either a bug or an unnecessary loop.

Reply:
It is a bug, also present in the original code. One of the iVar should be jVar. 

The question is whether this treatment of negative square of the speed of sound (setting the flux to zero) is the best one possible (if there already is a correct treatment). Maybe a dissipation term based on a^* would be better in this case (if it is only used during the iteration process).

Reply:
Agreed. Would clamping RoeSoundSpeed2 to some safe lower bound be a solution? Setting Jacobian contributions to 0 may make it ill-conditioned.

Reply:
At least better than what is currently in there, although these tricks usually only postpone the crash.